backend/app/api/threads.py

```diff
@@ -75,6 +82,26 @@ async def add_thread_state(
     )
 
 
+@router.patch("/{tid}/state")
+async def patch_thread_state(
+    user: AuthedUser,
+    tid: ThreadID,
+    payload: ThreadPatchRequest,
+):
+    """Patch state for a thread."""
+    thread = await storage.get_thread(user["user_id"], tid)
+    if not thread:
+        raise HTTPException(status_code=404, detail="Thread not found")
+    assistant = await storage.get_assistant(user["user_id"], thread["assistant_id"])
+    if not assistant:
```

Comment:
This check is not needed. We want to support patching thread state metadata even if the assistant has been deleted.